### ISSUE TITLE
bug: _check_version always returns False for <= and < operators; Seq2Seq teacher forcing is dead code

### ISSUE THREAD

**Issue by phoneee** [PyThaiNLP/pythainlp#1378]
### Description

Two latent bugs that do not currently affect normal usage but will cause problems under specific conditions:

1. **`_check_version()` in `corpus/core.py`** — The `<=` and `<` branches use `temp[0]` which takes the first **character** of the version string instead of the full string. `_check_version("<=5.3.3")` returns `False` even though the current version is `5.3.3`. No current corpus triggers a wrong result, but any future corpus using `<=5.x.x` will be wrongly rejected by `download()`.

2. **`Seq2Seq.forward()` in `transliterate/thai2rom.py`** — The teacher forcing if-else block is immediately overwritten by an unconditional `decoder_input = topi.detach()`. The shipped pre-trained model works fine, but retraining or fine-tuning would have teacher forcing silently disabled.

### Expected results

```python
from pythainlp.corpus.core import _check_version
_check_version("<=5.3.3")  # True
_check_version("<5.3.4")   # True
```

### Current results

```python
_check_version("<=5.3.3")  # False — WRONG
_check_version("<5.3.4")   # False — WRONG
```

### Steps to reproduce

```python
from pythainlp.corpus.core import _check_version
print(_check_version("<=5.3.3"))  # False (should be True)
print(_check_version("<9.9.9"))   # False (should be True)

import inspect
from pythainlp.transliterate.thai2rom import Seq2Seq
# Lines 421-426: if-else overwritten by unconditional assignment
```

### PyThaiNLP version

5.3.3

### Python version

3.13

### Operating system and version

macOS

### Possible solution

- `_check_version`: Change `temp[0]` to `temp` on lines 635, 638. Present since commit f9ead56f (2020-12-18).
- Teacher forcing: Remove unconditional `decoder_input = topi.detach()` on line 426. Left behind by commit d48dc503 (incomplete revert).

### Files

- `pythainlp/corpus/core.py` (lines 635, 638)
- `pythainlp/transliterate/thai2rom.py` (line 426)

**Comment by phoneee** [comment]
Closing this issue — it combined two unrelated bugs in one issue. Splitting into separate issues for clarity:

1. `_check_version` temp[0] → will open a new issue
2. Teacher forcing dead code → will open a new issue
